shared/tests/utility/support/TestYamlConversion.cpp

```diff
@@ -0,0 +1,201 @@
+#include <string>
+#include <utility>
+#include <yaml-cpp/yaml.h>
+
+#include "utility/support/yaml_expression.hpp"
+
+
+using utility::support::Expression;
+using utility::support::resolve_expression;
+
+/**
+ * Unit tests for [utility][support][yaml_expression]
+ *
+ * @author Liam Craft
+ */
+
+
+SCENARIO("yaml nodes can be converted to a given container type", "[utility][support][yaml_expression]") {
+    GIVEN("A yaml node containing a list of data") {
+        YAML::Node testValues = YAML::LoadFile("/home/nubots/NUbots/shared/tests/data/yamlConversionTestVals.yaml");
+        // error threshold when comparing doubles
+        static constexpr double ERROR_THRESHOLD = 1e-6;
+        // ints
+        WHEN("A vector is declared using integers from the YAML node") {
+            std::vector<int> intVector = resolve_expression<int>(testValues["ints"]);
+            // Check that the values are good(container size, data type....)
+            THEN("Container sizes should be equivalent") {
+                REQUIRE(testValues["ints"].size() == intVector.size());
+            }
+            AND_THEN("Values at each index should be equivalent") {
+                // Check values at each index
+                for (size_t i = 0; i < testValues["ints"].size(); i++) {
+                    REQUIRE(testValues["ints"][i].as<int>() == intVector.at(i));
+                }
+            }
+        }
+
+        // float
+        AND_WHEN("A vector is declared using float values from the YAML node") {
+            std::vector<float> floatVector = resolve_expression<float>(testValues["floats"]);
+            THEN("Container sizes should be equivalent") {
+                REQUIRE(testValues["floats"].size() == floatVector.size());
+            }
+            AND_THEN("Values at each index should be equivalent") {
+                for (size_t i = 0; i < testValues["floats"].size(); i++) {
+                    REQUIRE(testValues["floats"][i].as<float>() == floatVector.at(i));
+                }
+            }
+        }
+
+        // double
+        AND_WHEN("A vector is declared using doubles from the YAML node") {
+            std::vector<double> doubleVector = resolve_expression<double>(testValues["doubles"]);
+            THEN("Container sizes should be equivalent") {
+                REQUIRE(testValues["doubles"].size() == doubleVector.size());
+            }
+            AND_THEN("Values at each index should be equivalent") {
+                for (size_t i = 0; i < testValues["doubles"].size(); i++) {
+                    Approx epsilon_result = Approx(doubleVector.at(i)).epsilon(ERROR_THRESHOLD);
+                    REQUIRE(testValues["doubles"][i].as<double>() == epsilon_result);
+                }
+            }
+        }
+
+        // Vector2d
+        AND_WHEN("A vector is declared using Vector2d values from the YAML node") {
+            std::vector<Eigen::Vector2d> eigenV2dVector = resolve_expression<Eigen::Vector2d>(testValues["vector2"]);
```

Comment:
`snake_case` for variables and functions please, unless you're editing a file which is consistently camelCase
```suggestion
            std::vector<Eigen::Vector2d> eigen_vec2d_vector = resolve_expression<Eigen::Vector2d>(testValues["vector2"]);
```